### PR TITLE
Add JavaInteropSpeedBench.

### PR DESCRIPTION
**Pull request by zapster**
JMH micro benchmark derived from `JavaInteropSpeedTest` (cf graalvm/truffle#112). The test can be executed as follows:
```
mx microbench -- com.oracle.graal.truffle.bench.interop.JavaInteropSpeedBench
```

### CODE REVIEW

File: graal/com.oracle.graal.truffle.bench/src/com/oracle/graal/truffle/bench/interop/JavaInteropSpeedBench.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+@Fork(1)
+@State(Scope.Thread)
+public class JavaInteropSpeedBench {
+    private static final int REPEAT = 10000;
+    private static final long SEED = 42;
+    private static int[] arr;
+
+    @Setup
+    public static void beforeTesting() {
+        InstrumentationTestMode.set(true);
```
Comment (woess) [inline graal/com.oracle.graal.truffle.bench/src/com/oracle/graal/truffle/bench/interop/JavaInteropSpeedBench.java]: I believe we don't need `InstrumentationTestMode.set` anymore. Which means we should also remove it from the tests in the truffle repo. Maybe @chumer or @mlvdv can comment on that?

File: graal/com.oracle.graal.truffle.bench/src/com/oracle/graal/truffle/bench/interop/JavaInteropSpeedBench.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    @Benchmark
+    public int doMinMaxInJava() {
+        int max = 0;
+        for (int i = 0; i < arr.length; i++) {
+            max = Math.max(arr[i], max);
+        }
+        return max;
+    }
+
+    private static final TruffleObject TRUFFLE_MAX = JavaInterop.asTruffleFunction(IntBinaryOperation.class, new IntBinaryOperation() {
```
Comment (woess) [inline graal/com.oracle.graal.truffle.bench/src/com/oracle/graal/truffle/bench/interop/JavaInteropSpeedBench.java]: I'd suggest to use `java.lang.function.IntBinaryOperator` here and drop the `IntBinaryOperation` class.

**Comment by zapster** [comment]
Update: I've removed `InstrumentationTestMode` since we think we don't need it anymore (true? @chumer or @mlvdv) and it contained compilation errors.

**Comment by woess** [comment]
looks good to me, but waiting for @jtulach's approval

**Comment by chumer** [comment]
@zapster if it runs for you without InstrumentationTestMode you should be fine.

**Comment by zapster** [comment]
@chumer thanks for the info

**Pull request merged**